cmd/sync/sync.go

```diff
@@ -249,22 +288,88 @@ func syncRepository(dir string, entry SyncConfigEntry) error {
 			return err
 		}
 
-		if err := run(newGitCmd("merge", "--no-ff", "--no-commit", b.UpstreamLocalBranch())); err != nil {
-			if exitError, ok := err.(*exec.ExitError); ok {
-				fmt.Printf("---- Merge hit an ExitError: '%v'. A non-zero exit code is expected if there were conflicts. The script will try to resolve them, next.\n", exitError)
-			} else {
-				// Make sure we don't ignore more than we intended.
+		c := changedBranch{Refs: b}
+		var commitMessage string
+
+		if entry.SubmoduleTarget == "" {
```

Comment:
Not quite following this part. When there is no submodule target what are we merging here. Is this code flow between branches in the repo?

Reply:
Yeah, this path handles code flow and merging from an upstream repo into a fork. (`go-images` is still a fork for now. I mentioned earlier that I expected the submodule work to break `go-images`, but it didn't end up being difficult to preserve fork merging, so it will probably still work after the switch.)